### PR TITLE
Adding tests for AbstractArrayMath.jl

### PR DESCRIPTION
**Pull request by Priynsh**
added tests for lines 7, 137-154 (insertdims function) from base/abstractarraymath.jl

### CODE REVIEW

**Comment by nsajko** [comment]
Probably want a more descriptive PR title and commit message.

File: test/abstractarray.jl [hunk 1]
```diff
@@ -2206,3 +2206,44 @@ end
...
+        @test isreal(C) == true
+        D = Real[]
+        @test isreal(D) == true
+        E = [1 + 1im, 2 - 2im]
+        @test isreal(E) == false
+        struct MyReal <: Real
+            value::Float64
+        end
+        Base.convert(::Type{Float64}, x::MyReal) = x.value
+        Base.promote_rule(::Type{Float64}, ::Type{MyReal}) = Float64
```
Comment (LilithHafner) [inline test/abstractarray.jl:2225]:
```suggestion
```
The test is stricter if we don't define these. The `<: Real` annotation should be all we need.

File: test/abstractarray.jl [hunk 2]
```diff
@@ -2206,3 +2206,44 @@ end
...
+        F = [MyReal(1.0), MyReal(2.0)]
+        @test isreal(F) == true
+        G = ["a", "b", "c"]
+        @test_throws MethodError isreal(G)
+    end
+
+    @testset "insertdims tests" begin
```
Comment (LilithHafner) [inline test/abstractarray.jl:2232]:
```suggestion
    @testset "insertdims" begin
```
The word "test" is redundant here.

File: test/abstractarray.jl [hunk 3]
```diff
@@ -2206,3 +2206,44 @@ end
...
+        @test isreal(E) == false
+        struct MyReal <: Real
+            value::Float64
+        end
+        Base.convert(::Type{Float64}, x::MyReal) = x.value
+        Base.promote_rule(::Type{Float64}, ::Type{MyReal}) = Float64
+        F = [MyReal(1.0), MyReal(2.0)]
+        @test isreal(F) == true
+        G = ["a", "b", "c"]
+        @test_throws MethodError isreal(G)
```
Comment (LilithHafner) [inline test/abstractarray.jl:2229]: I appreciate you testing this failure case as well.

**Review by LilithHafner — COMMENTED** [review]
Thanks for contributing tests!

I can confirm that [codecov](https://app.codecov.io/gh/JuliaLang/julia/blob/master/base%2Fabstractarraymath.jl) reports both this `isreal` method and all of `insertdims` as untested. I don't see anywhere in `test/` where this `isreal` method is tested but I do see that `insertdims` is tested here:

https://github.com/JuliaLang/julia/blob/6cb9f04c1ec562f800669ba13a9a096203aea7cf/test/arrayops.jl#L312-L338

Codecov does not currently display [coverage results of this PR](https://app.codecov.io/gh/JuliaLang/julia/pull/56773).

**Comment by Priynsh** [comment]
thanks for such a detailed feedback! It really helps :) I had missed the insertdims test which were already present in the code, but they are not complete. So I have removed any redundancies for the same. Should i move all the insertdims test be together in arrayops? or is this okay?

**Comment by Priynsh** [comment]
I have added my tests in the same place as well. making a test-set for the same :)

**Review by LilithHafner — APPROVED** [review]
Thanks!

**Pull request merged**